$OpenBSD: patch-kamera_kcontrol_kameradevice_cpp,v 1.1 2013/01/19 11:28:00 zhuk Exp $
Fix compilation with GPhoto 2.5. Based on ideas from those patches:
http://bugsfiles.kde.org/attachment.cgi?id=73176
http://www.mail-archive.com/opensuse-commit@opensuse.org/msg30153.html
--- kamera/kcontrol/kameradevice.cpp.orig	Sat Jan 19 01:08:58 2013
+++ kamera/kcontrol/kameradevice.cpp	Sat Jan 19 01:11:01 2013
@@ -346,8 +346,10 @@ KameraDeviceSelectDialog::KameraDeviceSelectDialog(QWi
 	}
 	for (int i = 0; i < gphoto_ports; i++) {
 		if (gp_port_info_list_get_info(list, i, &info) >= 0) {
-			if (strncmp(info.path, "serial:", 7) == 0)
-				m_serialPortCombo->insertItem(QString::fromLatin1(info.path).mid(7));
+			char *path;
+			gp_port_info_get_name(info, &path);
+			if (strncmp(path, "serial:", 7) == 0)
+				m_serialPortCombo->insertItem(QString::fromLatin1(path).mid(7));
 		}
 	}
 	gp_port_info_list_free(list);
